can/interfaces/ics_neovi/neovi_bus.py

```diff
@@ -149,9 +149,13 @@ def _detect_available_configs():
         if ics is None:
             return []
         # TODO: add the channel(s)
-        return [{
-            'serial': NeoViBus.get_serial_number(device)
-        } for device in ics.find_devices()]
+        try:
+            return [{
+                'serial': NeoViBus.get_serial_number(device)
+            } for device in ics.find_devices()]
+        except Exception as e:
```

Comment:
I wonder if this `try`/`for` should be inverted - if the `try` was around each call to `get_serial_number` then any working devices can be returned?

Reply:
I made the code a bit clearer to show that only the call to `ics.find_devices()` is wrapped in a try-catch clause. The method `NeoViBus.get_serial_number()` itself should never fail.